_posts/2020/11/2020-11-23-python-repository.md

```diff
@@ -0,0 +1,140 @@
+python3 -m pip install --index-url https://central.artipie.com/olenagerasimova/pypi/ pypiexample
+python3 -m pypiexample
+``` 
+
+I hope that you see a polar plot with three curves, visualisation of our data files. 
+
+To publish package to the Artipie Central repository build it with `sdist` and use `twine` for upload:
+
+```commandline
+python setup.py sdist bdist_wheel
+twine upload --repository-url https://central.artipie.com/olenagerasimova/pypi -u olenagerasimova -p *** dist/*
+```
+
+I've set up [GitHub action](https://github.com/artipie/pypi-example/runs/1449528914?check_suite_focus=true) 
+to upload the example to Artipie Central on release, check the script 
+[here](https://github.com/artipie/pypi-example/blob/master/.github/workflows/python-publish.yml).
+
+To sum up, in this chapter we set up PyPi repository in Artipie Central, created sample Python 
+project, published and installed it. Now we'll move on to the binary storage and 
+I'll show where data files for the example are stored.
+
+## Binary Storage
+
+Artipie can be used as a [storage for any files](https://github.com/artipie/artipie/tree/master/examples/binary), 
+this repository is used to store experimental data which are visualised by the Python example. I've created 
+binary `data` repository on Central, exact configuration is the following:
+```yaml
+repo:
+  type: file
+  storage: default
+  permissions:
+    olenagerasimova:
+      - upload
+      - download
+    "*":
+      - download
+```
+
+As in the previous example I've allowed download operation to anyone. Data files are three text 
+files with some numbers, here is how the first one looks like:
+```text
+6
+3.5
+5
+4
+4.5
+3
+2.7
+5
+6
+3
+1.2
+3.2
+6
+```
+The other two files have the same form, only numbers are different. These files can be uploaded to 
+the Artipie Central `data` repository with simple `PUT` http request:
+
+```commandline
+httpie -a olenagerasimova:*** PUT https://central.artipie.com/olenagerasimova/data/y1.dat @data/y1.dat
+httpie -a olenagerasimova:*** PUT https://central.artipie.com/olenagerasimova/data/y2.dat @data/y2.dat
+httpie -a olenagerasimova:*** PUT https://central.artipie.com/olenagerasimova/data/y3.dat @data/y3.dat
+```
+
+To download the files use `GET` request
+
+```commandline
+httpie -a https://central.artipie.com/olenagerasimova/data/y1.dat > ./data/y1.dat
+``` 
+
+or simply open links [one](https://central.artipie.com/olenagerasimova/data/y1.dat), 
+[two](https://central.artipie.com/olenagerasimova/data/y2.dat) and 
+[three](https://central.artipie.com/olenagerasimova/data/y3.dat) in the browser. 
+The Python example downloads these files and visualise them on the polar plot.
+
+As this binary repository API is very simple (http `PUT` and `GET` requests), it's easy to write 
+a piece of code in any language to upload and downloads necessary files. What is also important, 
+authorisation is supported in Artipie, so it's possible to grant rights to upload/download from 
```

Comment:
@olenagerasimova I'd suggest to spell "authorization", see https://english.stackexchange.com/questions/282621/authorization-vs-authorisation-im-in-some-real-dilemma